### DISCUSSION TITLE
Is Codepipeline V2 supported ?

### DISCUSSION THREAD

**Discussion by VaZark** [aws/copilot-cli#5579] [Q&A] [answered]
I'd like to configure a pipeline that triggers when I add a new tag to my GitHub repository.

This is useful for versioned releases as any push/merge is deployed immediately to QA while production can be triggered only when we add a new tag

**Comment by KollaAdithya** [comment, marked as answer]
Hello @VaZark !

Copilot by default creates a code pipeline V1. But you can use `yamlPatches` to update the current pipeline to V2.

1. Run `copilot pipeline override --tool yamlpatch` and add the below patch to update the pipeline
```
- op: add
  path: /Resources/Pipeline/Properties/PipelineType
  value: V2
 ```
 2. Run `copilot pipeline update` to change pipelineType to V2.

**Reply by tyoc213** [reply]
if I have an already defined pipeline, the changue will update it? or do I need to delete it first and redo it again with this?

**Reply by KollaAdithya** [reply]
Hello @tyoc213 

You can update the already existing pipeline with above yamlpatch.
No need to delete the pipeline.
